submodules/eks/node-group.tf

```diff
@@ -73,21 +71,12 @@ locals {
   node_groups_by_name = { for ngz in local.node_groups_per_zone : "${ngz.ng_name}-${ngz.sb_name}" => ngz }
 }
 
-data "aws_ec2_instance_type" "all" {
-  for_each      = toset(flatten([for ng in local.node_groups : ng.instance_types]))
-  instance_type = each.value
-}
-
-locals {
-  any_gpu = { for name, ng in local.node_groups : name => anytrue([for itype in ng.instance_types : length(data.aws_ec2_instance_type.all[itype].gpus) > 0]) }
-}
```

Comment:
Shoving this out of the module is basically the fix for the node group recreation issue, but it means that we don't get autodetection in the module itself

Reply:
can you elaborate on what exactly triggers it

Reply:
Sure, it's mostly described in the ticket; essentially a change to the global tags will trigger a complete recreation of the node groups instead of a minor update to the tags themselves. This seems to be because the `aws_ec2_instance_type` data source is only "known during apply" which means that the `ami_type` for each node group is also deferred until apply and that triggers Terraform's complete recreate policy. I tried a few ways to keep the detection in-module, but ultimately only passing it in triggers the read early enough that it allows the minor updates.

Reply:
I knew what was triggering it but not the specifics. Getting data resources to resolve during plan is ideal but not always possible. My guess is that the subnet bits deferred everything to apply.